include/deal.II/base/aligned_vector.h

```diff
@@ -22,7 +22,15 @@
 #include <deal.II/base/memory_consumption.h>
 #include <deal.II/base/utilities.h>
 #include <deal.II/base/parallel.h>
-#include <boost/serialization/array.hpp>
+
+// boost::serialization::make_array used to be in array.hpp, but was
+// moved to a different file in BOOST 1.64
+#include <boost/version.hpp>
+#if BOOST_VERSION >= 106400
```

Comment:
You can't ask for `== 106400` because there may have been a boost 1.64.1 package that would show up as `106401`. If you want to go that route, you need `(BOOST_VERSION >= 106400) && (BOOST_VERSION < 106500)`. But I think @masterleinad thinks that the correct way of doing things is to assume that their long-term strategy is to stick with the `array_wrapper.hpp` header, and that we should just include that one. That makes sense to me -- in fact, if that's what we think they're going to do, we could think of just dropping `array.hpp` for newer BOOST versions altogether.